### PR TITLE
Add environment file serializer

### LINKED ISSUE #62983: [FEATURE REQUEST] environment file serializer
**Describe the solution you'd like**
A common workflow is to create environment files, which have an `=` between a key and value. It would be nice to have a serializer to handle that formatting.

**Describe alternatives you've considered**
`file.keyvalue` can perform a similar purpose, but it can't completely overwrite the file without invoking another state... and there also may be some value in having the functionality in a serializer for other non-state invocations.

**Additional context**
Format:
```
FOO=bar
```

**Please Note**
If this feature request would be considered a substantial change or addition, this should go through a SEP process here https://github.com/saltstack/salt-enhancement-proposals, instead of a feature request.

### PR DESCRIPTION
**Pull request by nicholasmhughes**
### What does this PR do?
Add an environment file serializer

### What issues does this PR fix or reference?
Fixes: #62983

### New Behavior
`file.serialize` can output a dataset such as:

```
foo: bar
wang: chung
```

or

```
- [foo, bar]
- [wang, chung]
```

as

```
foo=bar
wang=chung
```

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [x] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Comment by OrangeDog** [comment]
I think `properties` or `keyvalue` would be better names than `envfile`.

**Review by cmcmarrow — CHANGES_REQUESTED** [review]
Look like `assert serialized == "foo=bar baz\nsalt=rocks", serialized` will lead to test failer on windows because /r/n

**Comment by OrangeDog** [comment]
> Look like `assert serialized == "foo=bar baz\nsalt=rocks", serialized` will lead to test failer on windows because /r/n

Perhaps there should also be an option to choose line endings?

The options that are there do need to be documented though.

**Comment by nicholasmhughes** [comment]
re-run pr-freebsd-131-amd64-py3-pytest

**Review by cmcmarrow — APPROVED** [review]

**Review by cmcmarrow — CHANGES_REQUESTED** [review]
I agree with orange dog that /r/n should be an option and documented

File: salt/serializers/keyvalue.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
+"""
+    salt.serializers.keyvalue
```
Comment (Ch3LL) [inline salt/serializers/keyvalue.py:2]: can you add a versionadded for 3006

File: salt/serializers/keyvalue.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    Options:
+
+    :param line_ending:
+        String representation of LF or CRLF to be used for serialization to a
+        file. Defaults to ``\\r\\n`` on Windows and ``\\n`` on other operating
+        systems.
+
+    :param quoting:
+        Boolean flag to determine if values should be quoted (``True``) during
+        serialization or dequoted (``False``) during deserialization.
```
Comment (OrangeDog) [inline salt/serializers/keyvalue.py:18]: This doesn't seem right. Surely when `True` it should both quote when serializing and dequote when deserializing?

When `False` it should write and read raw values.

**Review by Ch3LL — CHANGES_REQUESTED** [review]

File: salt/serializers/keyvalue.py (hunk 2 above)
Comment (OrangeDog) [reply salt/serializers/keyvalue.py:18]: It also doesn't say what the default is.

File: salt/serializers/keyvalue.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+    :param separator:
+        String representing the character(s) used when concatenating or reading
```
Comment (OrangeDog) [inline salt/serializers/keyvalue.py:21]: Presumably you mean the separator between the key and the value?

It also doesn't say what the default is.

**Review by Ch3LL — APPROVED** [review]

File: salt/serializers/keyvalue.py (hunk 2 above)
Comment (OrangeDog) [reply salt/serializers/keyvalue.py]: According to the tests it also uses single quotes. There should probably be an option for double quotes.

**Comment by nicholasmhughes** [comment]
re-run pr-windows-2019-x64-py3-pytest

**Review by cmcmarrow — APPROVED** [review]

**Pull request merged**